selectors/views/channel_sidebar.ts

```diff
@@ -84,9 +84,11 @@ export const getUnreadChannels = (() => {
         (state: GlobalState) => getUnreadChannelIds(state),
         getCurrentChannelId,
         (allChannels, unreadChannelIds, currentChannelId) => {
-            const unreadChannels = unreadChannelIds.map((channelId) => allChannels[channelId]);
+            const unreadChannels = unreadChannelIds.
+                map((channelId) => allChannels[channelId]).
+                filter((channel) => channel.delete_at === 0);
```

Comment:
you can do it in just one go, to prevent going through the list multiple times:

```suggestion
            const unreadChannels = unreadChannelIds.
                reduce((result, channelId) => {
                    const channel = allChannels[channelId];

                    // remove archived channels from the list unless it is the one we are currently reading
                    if (channel.delete_at !== 0 || channel.id === currentChannelId) {
                        result.push(channel);
                    }
                    return result;
                 });
```

you can also use a foreach if you think it  would be more clear.

Reply:
I'm curious what the actual performance difference is here. I tend to use multiple calls for clarity, but that might be because I'm thinking too much about the big-O notation performance where looping through an array once is treated the same as doing it 100,000 times 😛.

I think I'll go for a for loop. I always have trouble wrapping my head around reduce statements for some reason, and I've heard for-in is substantially faster than forEach

Reply:
yes it is the same in big-O notation :) and most likely there is a negligible performance gain, but I guess my functional programmer ODT sees a map+filter and needs to convert it to a reduce 😅